### ISSUE TITLE
[cygnus-common][api] Cygnus is not forwarding Fiware-Service and Fiware-ServicePath headers to Orion

### ISSUE THREAD

**Issue by AnotherCodeArtist** [telefonicaid/fiware-cygnus#1124]
I just managed to bring Cygnus up and running as a docker container (using cygnus release/1.2.0) and started to play around with it. There seems to be a severe problem with service names and service paths (or is there anything in the configuration to enable multi-tenancy?).

How Orion reacts to these headers: Whenever you create an entity (or subscription) in orion and use a Fiware-Service header field in this request, orion will put the new document in a database with the name "orion-[fiware-service]". So it keeps data for different tenants in different databases, which IMHO is a nice idea.

But whenever I use these headers in subscription requests to Cygnus, the new subscription ends up in Orion's default database ("orion") and its service-path is set to "/#".

Here's what I did:

# POST-Request:
## Headers:
```
Fiware-Service: tourguide
Fiware-ServicePath: /sample
accept: application/json
content-type: application/json
```

## Payload:
```json
{
           "subscription":{
                   "description": "One subscription to rule them all",
                   "subject": {
                           "entities": [
                                   {
                                           "idPattern": ".*",
                                           "type": "Room"
                                   }
                           ],
                           "condition": {
                                   "attrs": [
                                          "temperature"
                                   ],
                                   "expression": {
                                           "q": "temperature>40"
                                   }
                           }
                   },
                   "notification": {
                          "http": {
                                  "url": "http://localhost:1234"
                           },
                           "attrs": [
                                   "temperature",
                                   "humidity"
                           ]
                   },
                   "expires": "2016-05-05T14:00:00.00Z","throttling": 5
           },
           "endpoint":{
                   "host":"orion",
                   "port":"1026",
                   "ssl":"false"
           }
   }
```
BTW: The current docu does not mention that an endpoint is necessary, at least most of the samples don't use it.

BTW: What was the idea behind writing this in the docs (Did you ever come across a GET request with a request body? For me this was the first time):
```json
GET http://<cygnus_host>:<management_port>/v1/subscriptions?ngsi_version=<ngsiVersion>&subscription_id=<subscriptionId>
   {
           "host":"orion.lab.fi-ware.org",
           "port":"1026",
           "ssl":"false",
           "xauthtoken":"QsENv67AJj7blC2rt54ESc5hMWYrs"
   }
```

# Resulting Document

So the expected behavior is that this entry ends up in mongodb in database "orion-tourguide" but instead the following document was written to the default ("orion") database:
```json
{
    "_id" : ObjectId("57ac789a1178d496caf06cf8"),
    "expiration" : NumberLong(1462456800),
    "reference" : "http://localhost:1234",
    "custom" : false,
    "throttling" : NumberLong(5),
    "servicePath" : "/#",
    "description" : "One subscription to rule them all",
    "status" : "active",
    "entities" : [ 
        {
            "id" : ".*",
            "type" : "Room",
            "isPattern" : "true"
        }
    ],
    "attrs" : [ 
        "temperature", 
        "humidity"
    ],
    "blacklist" : false,
    "conditions" : [ 
        {
            "type" : "ONCHANGE",
            "value" : [ 
                "temperature"
            ]
        }
    ],
    "expression" : {
        "q" : "temperature>40",
        "geometry" : "",
        "coords" : "",
        "georel" : ""
    },
    "format" : "normalized"
}
```

So you see, that also the service-path was changed.

**Comment by frbattid** [comment]
Nice catch! You are right, the [`handlePostSubscription`](https://github.com/telefonicaid/fiware-cygnus/blob/master/cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java#L882) method does not pass the FIWARE service nor the FIWARE service path at any time to the Orion backend class representing the Orion API:

* At orion backend creation time, only the host, port and ssl flag are passed.
* At orion backend usage, by invoking the `subscribeContextV1`/`subscribeContextV2` method, only the string representation of the Json subscription and the OAuth2 token are passed.

This must be fixed by passing both the FIWARE service and sevice path in the invocation of the `subscribeContextV1/subscribeContextV2` method.

**Comment by pcoello25** [comment]
Implemented in PR #1129
